plugin/restore/mysql/mysql.go

```diff
@@ -77,45 +84,51 @@ func (r *Restore) RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, bi
 }
 
 // SwapPITRDatabase renames the pitr database to the target, and the original to the old database
-func (r *Restore) SwapPITRDatabase(ctx context.Context, database string, timestamp int64) error {
-	db, err := r.driver.GetDbConnection(ctx, "")
-	if err != nil {
-		return err
-	}
+// It returns the pitr and old database names after swap.
+// It performs the step 2 of the restore process.
+func (r *Restore) SwapPITRDatabase(ctx context.Context, database string, suffixTs int64) (pitrDatabaseName string, pitrOldDatabase string, err error) {
+	pitrDatabaseName = getPITRDatabaseName(database, suffixTs)
+	pitrOldDatabase = getPITROldDatabaseName(database, suffixTs)
 
-	// TODO(dragonly): Clean up the transactions, they do not have a clear semantic / are not necessary.
-	txn, err := db.BeginTx(ctx, nil)
+	db, err := r.driver.GetDbConnection(ctx, "")
 	if err != nil {
-		return err
-	}
-	defer txn.Rollback()
-
-	pitrOldDatabase := getSafeName(database, "old")
-	pitrDatabaseName := getPITRDatabaseName(database, timestamp)
-
-	if _, err := txn.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE `%s`", pitrOldDatabase)); err != nil {
-		return err
+		return
```

Comment:
Even if we use named return values, let's still call out the returned values explicitly. 

https://github.com/golang/go/wiki/CodeReviewComments#naked-returns
points to
https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

"Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values."